Sources/PackageRegistry/RegistryManager.swift

```diff
@@ -0,0 +1,296 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import PackageLoading
+import PackageModel
+
+import TSCBasic
+import TSCUtility
+
+import struct Foundation.URL
+import struct Foundation.URLComponents
+import struct Foundation.URLQueryItem
+
+import Dispatch
+
+public enum RegistryError: Error {
+    case invalidOperation
+    case invalidResponse
+    case invalidURL
+    case invalidChecksum(expected: String, actual: String)
+}
+
+fileprivate let userAgent = "SwiftPackageManager/\(ToolsVersion.currentToolsVersion)"
```

Comment:
needed? the HTTPClient already does this

Reply:
Nope, not anymore. Removed with 48c3a023fd97f6e0dc9112213ecaac38cc4407c6